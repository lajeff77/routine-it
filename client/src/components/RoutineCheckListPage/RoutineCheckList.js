import React, {useState} from "react";
import {Box, Checkbox, FormGroup, FormControlLabel, FormControl, FormLabel, Typography} from "@mui/material";

const RoutineCheckList = (props) => {
    const [checked, setChecked] = useState(() => {
        const initialCheckedState = {};
        props.tasks.forEach(task => {
            initialCheckedState[task.name] = task.complete;
        });
        return initialCheckedState;
    });

    const handleChange = (event) =>
    {
        console.log(`Set ${event.target.name} = ${event.target.checked}`);
        setChecked({...checked, [event.target.name]: event.target.checked});
        //make api call here
    }

    return(
        <Box>
            <Typography variant="h2" sx={{textAlign:"center"}}>{props.name}</Typography>
            <FormControl component="fieldset">
                <FormLabel component="legend">Tasks</FormLabel>
                <FormGroup>
                    {props.tasks.map((task) => (
                        <FormControlLabel
                            key={task.id}
                            control={<Checkbox checked={checked[task.name]} onChange={handleChange} name={task.name} />}
                            label={task.name} 
                        />
                    ))}
                </FormGroup>
            </FormControl>
        </Box>
    )
}

export default RoutineCheckList;