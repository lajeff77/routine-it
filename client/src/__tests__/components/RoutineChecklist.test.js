import { render } from "@testing-library/react";
import {expect, test} from '@jest/globals';
import React from "react";
import RoutineCheckList from "../../components/RoutineCheckListPage/RoutineCheckList";

const testProps = {
    name: "Low Effort Routine",
    tasks: [{
        "id": 1,
        "routineId": 1,
        "name": "Brush teeth",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": true,
        "complete": false
    },
    {
        "id": 2,
        "routineId": 1,
        "name": "Wash face",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": true,
        "complete": false
    }]
};

describe(RoutineCheckList.name, ()=>{
    test('RoutineCheckList matches snapshot', () => {
        const {container} = render(<RoutineCheckList name={testProps.name} tasks={testProps.tasks}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});