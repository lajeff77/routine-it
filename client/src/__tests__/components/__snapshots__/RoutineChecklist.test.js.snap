// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutineCheckList RoutineCheckList matches snapshot 1`] = `
<div
  class="MuiBox-root css-0"
>
  <h2
    class="MuiTypography-root MuiTypography-h2 css-icizyq-MuiTypography-root"
  >
    Low Effort Routine
  </h2>
  <fieldset
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <legend
      class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
    >
      Tasks
    </legend>
    <div
      class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            name="Brush teeth"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Brush teeth
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            name="Wash face"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Wash face
        </span>
      </label>
    </div>
  </fieldset>
</div>
`;
